### DISCUSSION TITLE
[@types/oidc-provider] does not export types such as Client or Interaction

### DISCUSSION THREAD

**Discussion by breis-sa** [DefinitelyTyped/DefinitelyTyped#67434] [Issues with a @types package]
The `oidc-provider` `Provider` class takes a configuration object for which several properties are callbacks that receive types such as `Client` and `Interaction` as parameters. While I do see these types in `@types/oidc-provider`, they're not exported preventing them from being used, forcing us use `any` in those callback signatures.

@panva maybe I'm missing something or using the type definitions incorrectly so feel free to correct me, but here's an example of what I mean:

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/133907865/b1f9a4d4-8775-42ef-802d-5496835aa931)

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/133907865/73843a4c-db13-4c48-a5d4-16e472dc8371)

[a comment by an automated account is not shown]

**Reply by panva** [reply]
Resolved in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69979
